WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift

```diff
@@ -37,13 +39,20 @@ final class ItemListViewModel: ItemListViewModelProtocol {
 
     @MainActor
     func populatePointOfSaleItems() async {
+        let nextPage = currentPage + 1
         do {
             state = .loading
-            items = try await itemProvider.providePointOfSaleItems(pageNumber: Constants.firstPageNumber)
+            let newItems = try await itemProvider.providePointOfSaleItems(pageNumber: nextPage)
+            let uniqueNewItems = newItems.filter { newItem in
+                !items.contains(where: { $0.productID == newItem.productID })
+            }
+            items.append(contentsOf: uniqueNewItems)
```

Comment:
💬 Another thought for optimization (please ignore if this is already a future task): if the `items` count is less than the page size, it's safe to assume we've reached the last page so that no new API requests are made.

Reply:
That sounds reasonable, specially for the current limit of 100 items, I've [added it](https://github.com/woocommerce/woocommerce-ios/issues/14186) to the main issue along other optimization items.

Reply:
> That sounds reasonable, specially for the current limit of 100 items, I found this not working well when we have a large gap of unsupported items, but that would be an edge case and less prone to happen as we increase the number of supported products.

If there are items filtered and hidden from the items in the API response, I imagine this check on whether there's a next page would be done on the items in the API response like in `POSProductProvider.providePointOfSaleItems` and then passed to the view model level here.